electrumsv/wallet_database/functions.py

```diff
@@ -2163,6 +2220,17 @@ def _insert_transaction(self, db: sqlite3.Connection, tx_row: TransactionRow,
             if e.args[0] == "UNIQUE constraint failed: Transactions.tx_hash":
                 raise TransactionAlreadyExistsError()
 
+        # No uniqueness constraint as there can be more than one proof for the same tx_hash
+        # (i.e. orphaned proofs) And there can be more than one transaction in a block
+        # (so block_hash is not unique either)
+        if tx_row.block_hash is not None and tx_row.block_position is not None \
```

Comment:
Are there some invalid cases we should error for here where some of these are set but not the right ones?

Reply:
Any transaction imports should either set all three of these fields or none at all (and it does currently). But I supposed in the name of writing "defensive code" it would make sense to enforce this - perhaps with a `IncompleteProofDataSubmittedError` raised